src/commands/cmd_json.cc

```diff
@@ -204,11 +204,62 @@ class CommandJsonArrLen : public Commander {
   }
 };
 
+class CommandJsonStrAppend : public Commander {
+ public:
+  Status Execute(Server *svr, Connection *conn, std::string *output) override {
+    redis::Json json(svr->storage, conn->GetNamespace());
+
+    // path default, if not provided
+    std::string path = "$";
+    if (args_.size() == 4) {
+      path = args_[2];
+    }
+    std::vector<uint64_t> results;
+    auto s = json.StrAppend(args_[1], path, args_[3], results);
+    if (!s.ok()) return {Status::RedisExecErr, s.ToString()};
+
+    *output = IntegerArray(results);
+    return Status::OK();
+  }
+
+  static std::string IntegerArray(const std::vector<uint64_t> &values, bool output_nil_for_negative = true) {
+    std::string result = "*" + std::to_string(values.size()) + CRLF;
+    for (const auto &value : values) {
+      if (value == (uint64_t)-1 and output_nil_for_negative) {
```

Comment:
`(uint64_t)-1` is NOT a negative number. You can just remove `output_nil_for_negative`.